stress/shell/src/main/java/alluxio/stress/cli/RegisterWorkerBench.java

```diff
@@ -0,0 +1,208 @@
+   *
+   * 3. TODO: Simulate contention over block lock DefaultBlockMaster.mBlockLocks
+   * */
+  private RpcTaskResult simulateRegisterWorker(alluxio.worker.block.BlockMasterClient client, Instant endTime) {
+    RpcTaskResult result = new RpcTaskResult();
+    long i = 0;
+
+    WorkerNetAddress address;
+    long workerId = -1;
+    String hostname = NetworkAddressUtils.getLocalHostName(500);
+    LOG.info("Detected local hostname {}", hostname);
+    // TODO(jiacheng): need this?
+    if (mParameters.mSameWorker) {
+      LOG.info("Simulating the same worker registering again.");
+      address = new WorkerNetAddress().setHost(hostname)
+              .setDataPort(mPortToAssign.getAndIncrement())
+              .setRpcPort(mPortToAssign.getAndIncrement());
+      try {
+        workerId = client.getId(address);
+        LOG.info("Got worker ID {}", workerId);
+      } catch (Exception e) {
+        LOG.error("Failed to run iter {}", i, e);
+        result.addError(e.getMessage());
+        return result;
+      }
+    } else {
+      LOG.info("Simulating a new worker each time");
+    }
+
+    // TODO(jiacheng): default?
+    if (mParameters.mOnce) {
+      LOG.info("Only run once");
+      // If specified to run once, ignore the time stamp and run once
+      runOnce(client, result, i, workerId, hostname);
+    } else {
+      // Otherwise, keep running for as many times as possible
+      while (Instant.now().isBefore(endTime)) {
+        runOnce(client, result, i, workerId, hostname);
+      }
+    }
+
+    return result;
+  }
+
+  private void runOnce(alluxio.worker.block.BlockMasterClient client, RpcTaskResult result, long i, long workerId, String hostname) {
+    WorkerNetAddress address;
+    // TODO(jiacheng): The 1st reported RPC time is always very long, this does
+    //  not match with the time recorded by Jaeger.
+    //  I suspect it's the time spend in establishing the connection.
+    //  The easiest way out is just to ignore the 1st point.
+    try {
+      if (!mParameters.mSameWorker) {
+        // If use different worker, get a different address and workerId every time
+        address = new WorkerNetAddress().setHost(hostname).setDataPort(mPortToAssign.getAndIncrement()).setRpcPort(mPortToAssign.getAndIncrement());
+        workerId = client.getId(address);
+        LOG.info("Got new worker ID {}", workerId);
+      }
+
+      List<String> tierAliases = new ArrayList<>();
+      tierAliases.add("MEM");
+      long cap = 20L * 1024 * 1024 * 1024; // 20GB
+      Map<String, Long> capMap = ImmutableMap.of("MEM", cap);
+      Map<String, Long> usedMap = ImmutableMap.of("MEM", 0L);
+
+      Instant s = Instant.now();
+      client.register(workerId,
+              tierAliases,
+              capMap,
+              usedMap,
+              // Will use the prepared block list instead of converting this one
+              ImmutableMap.of(),
```

Comment:
is this why we're subclassing the client?
do we need to avoid this? Won't this effect timings?

Reply:
The principles when we are designing the tests are:
1. Generate as much pressure on the master as possible, so theoretically the only thing that the client should do is to send RPC requests (instead of doing some preprocessing and preparation on the client side each run).
2. Make the measured number as close to the master-side effort as possible. When an RPC takes X millisecond for the round time trip, we'd like the client to NOT waste time generating data in the X millisecond and NOT deserialize the data, as much as possible. 
3. Also the client-side work will take CPU and memory. When the concurrency is high (many client threads on one node), the stress test may be CPU-bounded on the client side, which makes the results meaningless. We are manually validating this as well. For example, manually testing different client-side concurrency to find a good number which introduces enough stress on master and not introduce client-side CPU bound.

Back to this particular line, this is relevant to Principle 2. We want the client to NOT spend time on converting data when doing the RPC. Same applies to `PinListFileSystemMasterClient`.